monkeytype/cli.py

```diff
@@ -235,11 +236,38 @@ def main(argv: List[str], stdout: IO, stderr: IO) -> int:
 
     args = parser.parse_args(argv)
     update_args_from_config(args)
+
     handler = getattr(args, 'handler', None)
     if handler is None:
         parser.print_help(file=stderr)
         return 1
-    handler(args, stdout, stderr)
+
+    handler_exception: Optional[Exception] = None
+    try:
+        with args.config.cli_context(args.command):
+            # Capture the handler exception so it doesn't inadvertently trigger the cli_context() helper errors.
+            try:
+                handler(args, stdout, stderr)
+            except Exception as e:
+                handler_exception = e
+    except (AttributeError, RuntimeError, TypeError) as e:
```

Comment:
Again here, definitely appreciate the concern for helping people out when things go wrong, but I think some of the error messages here make assumptions about how `cli_context` must be implemented (with `contextlib.contextmanager`), and about the source of a given exception type, that aren't necessarily valid. `TypeError` or `AttributeError` within their code could have an entirely different source from the one you are anticipating. Fortunately thanks to `raise from` you don't lose the original traceback here, which is good, but my feeling is still that the benefit isn't worth the complexity here, and we're better off relying on Python's own exception messages to be clear enough.

I probably triggered your extra caution here by my previous comment mentioning concern that requiring a context manager from the API might be harder for some developers! I'd rather see if a good example in the docs is enough to avoid that problem, and if not, do more once we have real-world data on what causes confusion.